src/java/org/apache/cassandra/repair/autorepair/UnrepairedBytesBasedTokenRangeSplitter.java

```diff
@@ -0,0 +1,335 @@
+                    }
+                    repairAssignments.add(assignment);
+                    targetBytesSoFar += approximateUnrepairedBytesForRange;
+                }
+                else
+                {
+                    long targetRanges = approximateUnrepairedBytesForRange / subrangeBytes;
+                    // TODO: approximation per range, this is a bit lossy since targetRanges rounds down.
+                    long approximateBytesPerSplit = approximateUnrepairedBytesForRange / targetRanges;
+                    logger.info("Splitting {}.{} for range {} into {} sub ranges, approximateBytesPerSplit={}", keyspaceName, tableName, tokenRange, targetRanges, FileUtils.stringifyFileSize(approximateBytesPerSplit));
+                    List<Range<Token>> splitRanges = splitEvenly(tokenRange, (int) targetRanges);
+                    int splitRangeCount = 0;
+                    for (Range<Token> splitRange : splitRanges)
+                    {
+                        RepairAssignment assignment = new BytesBasedRepairAssignment(splitRange, keyspaceName, Collections.singleton(tableName), approximateBytesPerSplit);
+                        if (!canAddAssignment(assignment, targetBytesSoFar, approximateBytesPerSplit))
+                        {
+                            return repairAssignments;
+                        }
+                        logger.info("Added repair assignment for {}.{} for subrange {} (#{}/{}) with approximateBytes={}",
+                                    keyspaceName, tableName, splitRange, ++splitRangeCount, splitRanges.size(), FileUtils.stringifyFileSize(approximateBytesPerSplit));
+                        repairAssignments.add(assignment);
+                        targetBytesSoFar += approximateBytesPerSplit;
+                    }
+                }
+            }
+            else
+            {
+                ColumnFamilyStore cfs = ColumnFamilyStore.getIfExists(keyspaceName, tableName);
+                long memtableSize = cfs.getTracker().getView().getCurrentMemtable().getLiveDataSize();
+                if (memtableSize > 0L)
+                {
+                    logger.info("Included {}.{} range {}, had no unrepaired SSTables, but memtableSize={}, adding single repair assignment", keyspaceName, tableName, tokenRange, memtableSize);
+                    RepairAssignment assignment = new BytesBasedRepairAssignment(tokenRange, keyspaceName, Collections.singleton(tableName), memtableSize);
+                    if (targetBytesSoFar >= maxBytesPerScheduleBytes)
+                    {
+                        return repairAssignments;
+                    }
+                    repairAssignments.add(assignment);
+                    targetBytesSoFar += memtableSize;
+                }
+                else
+                {
+                    logger.info("Skipping {}.{} for range {} because it had no unrepaired SSTables and no memtable data", keyspaceName, tableName, tokenRange);
```

Comment:
```
INFO  [AutoRepair-Repair-incremental:1] 2024-09-25 19:11:58,314 UnrepairedBytesBasedTokenRangeSplitter.java:101 - Calculating token range splits for repairType=incremental primaryRangeOnly=true keyspaceName=easy_cass_stress tableNames=[keyvalue]
INFO  [AutoRepair-Repair-incremental:1] 2024-09-25 19:11:58,314 UnrepairedBytesBasedTokenRangeSplitter.java:124 - Calculating unrepaired bytes for easy_cass_stress.keyvalue for range (3074457345618258602,-9223372036854775808]
INFO  [AutoRepair-Repair-incremental:1] 2024-09-25 19:11:58,314 UnrepairedBytesBasedTokenRangeSplitter.java:165 - Skipping over easy_cass_stress.keyvalue /Users/tolbertam/.ccm/irtest/node1/data0/easy_cass_stress/keyvalue-2f0a26b079e911ef8554bf6dd991a408/nb-
1070-big (60.06 MiB) because it is repaired
INFO  [AutoRepair-Repair-incremental:1] 2024-09-25 19:11:58,314 UnrepairedBytesBasedTokenRangeSplitter.java:165 - Skipping over easy_cass_stress.keyvalue /Users/tolbertam/.ccm/irtest/node1/data0/easy_cass_stress/keyvalue-2f0a26b079e911ef8554bf6dd991a408/nb-
1047-big (64.31 MiB) because it is repaired
INFO  [AutoRepair-Repair-incremental:1] 2024-09-25 19:11:58,315 UnrepairedBytesBasedTokenRangeSplitter.java:165 - Skipping over easy_cass_stress.keyvalue /Users/tolbertam/.ccm/irtest/node1/data0/easy_cass_stress/keyvalue-2f0a26b079e911ef8554bf6dd991a408/nb-
973-big (139.55 MiB) because it is repaired
INFO  [AutoRepair-Repair-incremental:1] 2024-09-25 19:11:58,315 UnrepairedBytesBasedTokenRangeSplitter.java:165 - Skipping over easy_cass_stress.keyvalue /Users/tolbertam/.ccm/irtest/node1/data0/easy_cass_stress/keyvalue-2f0a26b079e911ef8554bf6dd991a408/nb-
281-big (182.94 MiB) because it is repaired
INFO  [AutoRepair-Repair-incremental:1] 2024-09-25 19:11:58,315 UnrepairedBytesBasedTokenRangeSplitter.java:234 - Skipping easy_cass_stress.keyvalue for range (3074457345618258602,-9223372036854775808] because it had no unrepaired SSTables and no memtable d
ata
```

Probably would want this to submit a repair anyways, or maybe make it a configurable option.